src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/DiagnosticCounter.cs

```diff
@@ -77,7 +78,7 @@ public void AddMetadata(string key, string value)
 
         public string? DisplayName { get; set; }
 
-        public string? DisplayUnits { get; set; }
+        public string DisplayUnits { get; set; }
```

Comment:
Should the setter validate that `value != null`?  Even though the property is `string` rather than `string?`, someone could still set it to null.

Reply:
Technically, someone could set it to null but nothing terrible would happen because when it is consumed to generate the counter payload, we check it for null and put an empty string in that case. I changed it to throw an ArgumentException though.